BookkeeperAPI/Controllers/ExpenseController.cs

```diff
@@ -3,24 +3,155 @@
+        private readonly BookkeeperContext _context;
+        public ExpenseController(BookkeeperContext context)
         {
             _context = context;
         }
 
-        [HttpGet]
-        public ActionResult<List<Expense>> GetExpense(Guid userId)
+        [HttpGet("/api/me/expenses")]
+        public async Task<ActionResult<List<ExpenseView>>> GetExpense(Guid userId)
         {
-            return Ok(_context.Expenses.Where(x => x.User.Id.Equals(userId)));
+            if(userId == Guid.Empty)
+            {
+                return BadRequest();
+            }
+
+            return Ok(
+                await _context.Expenses
+                .Where(x => x.UserId.Equals(userId))
+                .Select(x => new ExpenseView()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Amount = x.Amount,
+                    Category = x.Category.ToString(),
+                    Date = x.Date,
+                }).ToListAsync());
+        }
+
+        [HttpGet("/api/expenses/{expenseId}")]
+        public async Task<ActionResult<ExpenseView>> GetExpenseById(Guid userId, Guid expenseId)
+        {
+            if (userId == Guid.Empty)
+            {
+                return BadRequest();
+            }
+
+            return Ok(
+                await _context.Expenses
+                .Where(x => x.UserId.Equals(userId) && x.Id.Equals(expenseId))
+                .Select(x => new ExpenseView()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Amount = x.Amount,
+                    Category = x.Category.ToString(),
+                    Date = x.Date,
+                })
+                .FirstOrDefaultAsync());
+        }
+
+        [HttpPost("/api/expenses")]
+        public async Task<ActionResult<ExpenseView>> CreateExpense(Guid userId, CreateExpenseRequest expense)
+        {
+            if (userId == Guid.Empty)
+            {
+                return BadRequest();
+            }
+
+            Expense _expense = new Expense();
+            _expense.UserId = userId;
+            _expense.Name = expense.Name;
+            _expense.Amount = expense.Amount;
+            _expense.Category = expense.Category.ToEnum<ExpenseCategory>();
+            _expense.Date = expense.Date;
+
+            await _context.Expenses.AddAsync(_expense);
+
+            await _context.SaveChangesAsync();
+
+            ExpenseView result = new ExpenseView()
+            {
+                Id = _expense.Id,
+                Name = _expense.Name,
+                Category = _expense.Category.ToString(),
+                Amount = _expense.Amount,
+                Date = _expense.Date,
+            };
+
+            return Ok(result);
+        }
+
+        [HttpPatch("/api/expenses/{expenseId}")]
+        public async Task<ActionResult<ExpenseView>> UpdateExpense(Guid userId, Guid expenseId, UpdateExpenseRequest expense)
```

Comment:
Same here, unless one expense can belong to multiple users, there is no point. If you are taking userId for validation whether this expense belong to the user or not then take userId from the token.

Reply:
Since we haven't implemented authorization and authentication yet, I was taking user Id as a parameter, but we can always come back and implement validation here.